server/models/meshery_filter_persister.go

```diff
@@ -54,6 +56,49 @@ func (mfp *MesheryFilterPersister) GetMesheryFilters(search, order string, page,
 	return marshalMesheryFilterPage(mesheryFilterPage), nil
 }
 
+// GetMesheryCatalogFilters returns all of the public filters
+func (mfp *MesheryFilterPersister) GetMesheryCatalogFilters(search, order string) ([]byte, error) {
+	order = sanitizeOrderInput(order, []string{"created_at", "updated_at", "name"})
+
+	if order == "" {
+		order = "updated_at desc"
+	}
+
+	filters := []*MesheryFilter{}
+
+	query := mfp.DB.Where("visibility = 'public'").Order(order)
```

Comment:
We have the concept of public for local provider as well? We can lose the visibility filter because on Meshery instance level. I dont think we need to care about public and private.

Reply:
We would be storing with visibility as `private` for all in local provider so I think we can lose the `public` from here.